src/Tests/DcsBios/DCSBIOSControlLocatorTests.cs

```diff
@@ -0,0 +1,363 @@
+
+            DCSBIOSControlLocator.DCSAircraft = dcsAircraft;
+            DCSBIOSControlLocator.JSONDirectory = jsonPath;
+
+            var controls = DCSBIOSControlLocator.GetOutputControls(dcsBiosOutputType);
+            foreach (var dcsbiosControl in controls)
+            {
+                var count = dcsbiosControl.Outputs.Count(o => o.OutputDataType == dcsBiosOutputType);
+                Assert.True(count > 0);
+
+            }
+        }
+
+        [Theory]
+        [InlineData(@"%USERPROFILE%\Saved Games\DCS\Scripts\DCS-BIOS\doc\json", 5, DCSBiosOutputType.IntegerType)]
+        [InlineData(@"%USERPROFILE%\Saved Games\DCS\Scripts\DCS-BIOS\doc\json", 5, DCSBiosOutputType.StringType)]
+        public void TestLocatorGetOutputControlsDCSBIOSOutputEnabledFlagNotSet(string jsonPath, int dcsAircraftId, DCSBiosOutputType dcsBiosOutputType)
+        {
+            jsonPath = Environment.ExpandEnvironmentVariables(jsonPath);
+            if (!Directory.Exists(jsonPath))
+            {
+                //No need to test as dcs-bios isn't installed
+                return;
+            }
+            Common.ResetEmulationModesFlag();
+            // Since Flag DCSBIOSOutputEnabled is not set DCSBIOSControlLocator returns null.
+
+            DCSAircraft.FillModulesListFromDcsBios(jsonPath, true);
+            var dcsAircraft = DCSAircraft.GetAircraft(dcsAircraftId);
+
+            DCSBIOSControlLocator.DCSAircraft = dcsAircraft;
+            DCSBIOSControlLocator.JSONDirectory = jsonPath;
+
+            var controls = DCSBIOSControlLocator.GetOutputControls(dcsBiosOutputType);
+            Assert.Null(controls);
+        }
+
+        [Theory]
+        [InlineData(@"%USERPROFILE%\Saved Games\DCS\Scripts\DCS-BIOS\doc\json", 5)]
+        public void TestLocatorGetInputControls(string jsonPath, int dcsAircraftId)
+        {
+            jsonPath = Environment.ExpandEnvironmentVariables(jsonPath);
+            if (!Directory.Exists(jsonPath))
+            {
+                //No need to test as dcs-bios isn't installed
+                return;
+            }
+            Common.ResetEmulationModesFlag();
+            Common.SetEmulationModes(EmulationMode.DCSBIOSInputEnabled);
+
+            DCSAircraft.FillModulesListFromDcsBios(jsonPath, true);
+            var dcsAircraft = DCSAircraft.GetAircraft(dcsAircraftId);
+
+            DCSBIOSControlLocator.DCSAircraft = dcsAircraft;
+            DCSBIOSControlLocator.JSONDirectory = jsonPath;
+
+            var controls = DCSBIOSControlLocator.GetInputControls();
+            foreach (var dcsbiosControl in controls)
+            {
+                var count = dcsbiosControl.Inputs.Count;
+                Assert.True(count > 0);
+
+            }
+        }
+
+        [Theory]
+        [InlineData(@"%USERPROFILE%\Saved Games\DCS\Scripts\DCS-BIOS\doc\json", 5)]
+        public void TestLocatorGetOutputControlsDCSBIOSInputEnabledFlagNotSet(string jsonPath, int dcsAircraftId)
```

Comment:
Maybe if the is only one value, write a Fact test and not a theory but I understand that in this case, tests could be added later and there is a kind of "beauty" in the sense that all test in this fileare in theories  and it could ease the reading somehow.